### DISCUSSION TITLE
How does recovering from kernel disconnection work in thebe/binderhub stack

### DISCUSSION THREAD

**Discussion by jokuNimi** [executablebooks/meta#490] [Q&A]
**Question:** what is the current status of getting information about and/or recovering from the kernel dying? Specifically
1) Is there a way of restarting a binder notebook after losing connection completely?
3) What happens if I re-press the "launch thebe" button after a disconnection?
3) Is there a nice way of getting information about a disconnection, similar to the event hooks that give information about the launch process? Is there a way of easily showing this information to the user (e.g. to ask them to refresh the page)?

**Context:** I'm using sphinx-thebe to add interactive code to a website. This in turn uses thebe to request notebooks from a binderhub server. If I activate the code and wait for long enough, the kernel seems to disconnect permanently. Then pressing "Run" either does nothing or shows an error. Pressing the "launch thebe" button doesn't seem to fix the issue, even if the launch process succeeds. This can be quite frustrating for an end-user who doesn't know what's going on.

Thanks in advance to anyone who reads this, or even better, responds :). I tried to read the documentation, but didn't find relevant information. I'm open to developing new features that would improve the situation, if that would be helpful.

**Comment by chrisjsewell** [comment]
Cc’ing @stevejpurves who has being doing a lot of work on thebe
